client/packages/inventory/src/Stocktake/api/api.ts

```diff
@@ -225,11 +225,9 @@ const getInsertStocktakeLines = (
   items: { itemId: string; stockLines?: StockLineFragment[] | undefined }[]
 ) => {
   const insertStocktakeLines = [] as InsertStocktakeLineInput[];
-
   items.forEach(item => {
     const { itemId, stockLines } = item;
```

Comment:
So you either
- add lines from existing stocklines
- add line for an item that doesn't has an existing stockline

right?

can that better be express through:
``` typescript
type Item = {
  itemId?: string,
  stockLines?: StockLineFragment[],
}

// then 
if (item.stockLines) {...} else if (item.ItemId) {...}

// or even define it like
type Item = string | StockLineFragment[];
```

(had some trouble figure out the logic, maybe comment helps as well)

Reply:
yes, that's right - it creates stocktake lines for each stock line, if there are stocklines.
If no stocklines, you still want a placeholder stocktake line, so create one which references the item.

I've added the `Item` typing and added comments. Hope it is clearer!